### PR TITLE
DOCSUP-7086: Edit and translate to Russian (s3 and updating ClickHouse)

### PR DESCRIPTION
**Pull request by ka1bi4**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation for #19748 and #19338

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Added to RU documentation:
- S3 table engine and Server Side Encryption Customer Keys support in S3 client.
- Additional explanation for the ClickHouse version upgrade.

### CODE REVIEW

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+``` sql
+ENGINE = S3(path, [aws_access_key_id, aws_secret_access_key,] format, structure, [compression])
+```
+
+**Входные параметры**
+
+-   `path` — URL бакета, включая путь к файлу. В режиме только для чтения поддерживаются следующие подстановочные знаки: `*`, `?`, `{abc,def}` и `{N..M}`, где `N`, `M` — числа, `’abc’`, `‘def’` — строки.
+-   `format` — [формат](../../../interfaces/formats.md#formats) файла.
+-   `structure` — структура таблицы. Указывается в формате `'column1_name column1_type, column2_name column2_type, ...'`.
+-   `compression` — опциональный параметр, необходимый, если требуется использовать определенный формат сжатия. Поддерживаемые значения: `none`, `gzip/gz`, `brotli/br`, `xz/LZMA`, `zstd/zst`. По умолчанию формат сжатия определяется автоматически, исходя из расширения файла.
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:21]:
```suggestion
-   `compression` — необязательный параметр. Необходим если требуется использовать определенный формат сжатия. Поддерживаемые значения: `none`, `gzip/gz`, `brotli/br`, `xz/LZMA`, `zstd/zst`. По умолчанию формат сжатия определяется автоматически исходя из расширения файла.
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+**Примеры**
+
+**1.** Настройте таблицу `s3_engine_table`:
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:25]:
```suggestion
**1.** Создайте таблицу `s3_engine_table`:
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+``` sql
+CREATE TABLE s3_engine_table (name String, value UInt32) ENGINE=S3('https://storage.yandexcloud.net/my-test-bucket-768/test-data.csv.gz', 'CSV', 'name String, value UInt32', 'gzip');
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:28]:
```suggestion
CREATE TABLE s3_engine_table (name String, value UInt32) ENGINE = S3('https://storage.yandexcloud.net/my-test-bucket-768/test-data.csv.gz', 'CSV', 'name String, value UInt32', 'gzip');
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+└──────┴───────┘
+```
+
+## Детали реализации {#detali-realizatsii}
+
+-   поддерживается многопоточное чтение и запись.
+-   не поддерживается:
+    -   использование операций `ALTER` и `SELECT...SAMPLE`;
+    -   индексы;
+    -   репликация.
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:60]:
```suggestion
-  Поддерживается многопоточное чтение и запись.
-  Не поддерживается:
    -   использование операций `ALTER` и `SELECT...SAMPLE`;
    -   индексы;
    -   репликация.
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+**Шаблоны в пути**
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:62]:
```suggestion
**Символы подстановки в пути**
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+Шаблоны могут содержаться в нескольких компонентах пути. Обрабатываются только существующие файлы, название которых целиком удовлетворяет шаблону (не только суффиксом или префиксом). Список файлов определяется во время выполнения запроса `SELECT` (не во время `CREATE`).
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:64]:
```suggestion
Символы подстановки могут содержаться в нескольких компонентах пути. Обрабатываются только существующие файлы, название которых целиком удовлетворяет шаблону (не только суффиксом или префиксом). Список файлов определяется во время выполнения запроса `SELECT` (не во время `CREATE`).
```

File: docs/ru/engines/table-engines/integrations/s3.md (hunk 5 above)
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:62]:
```suggestion
**Символы подстановки в пути**
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+-   `*` — заменяет любое количество любых символов кроме `/`, включая отсутствие символов.
+-   `?` — заменяет любой одиночный символ.
+-   `{some_string,another_string,yet_another_one}` — заменяет любую из строк `'some_string', 'another_string', 'yet_another_one'`.
+-   `{N..M}` — заменяет любое число в интервале от `N` до `M` включительно (может содержать ведущие нули, например: `000..078`).
+
+Конструкция с `{}` аналогична табличной функции [remote](../../../sql-reference/table-functions/remote.md).
+
+**Пример**
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:73]:
```suggestion
**Примеры**
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+1. Предположим, что у нас есть несколько файлов в формате TSV со следующими URL-адресами в HDFS:
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:75]:
```suggestion
Предположим, что у нас есть несколько файлов в формате TSV со следующими URL-адресами в HDFS:
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+-   ‘https://storage.yandexcloud.net/my-test-bucket-768/some_prefix/some_file_1.csv’
+-   ‘https://storage.yandexcloud.net/my-test-bucket-768/some_prefix/some_file_2.csv’
+-   ‘https://storage.yandexcloud.net/my-test-bucket-768/some_prefix/some_file_3.csv’
+-   ‘https://storage.yandexcloud.net/my-test-bucket-768/another_prefix/some_file_1.csv’
+-   ‘https://storage.yandexcloud.net/my-test-bucket-768/another_prefix/some_file_2.csv’
+-   ‘https://storage.yandexcloud.net/my-test-bucket-768/another_prefix/some_file_3.csv’
+
+2. Существует несколько способов создать таблицу, объединяющую все шесть файлов:
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:84]:
```suggestion
Существует несколько способов создать таблицу, объединяющую все шесть файлов:
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+
+``` sql
+CREATE TABLE table_with_range (name String, value UInt32) ENGINE = S3('https://storage.yandexcloud.net/my-test-bucket-768/{some,another}_prefix/some_file_{1..3}', 'CSV');
+```
+
+3. Еще один способ:
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:90]:
```suggestion
Второй способ:
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+
+``` sql
+CREATE TABLE table_with_question_mark (name String, value UInt32) ENGINE = S3('https://storage.yandexcloud.net/my-test-bucket-768/{some,another}_prefix/some_file_?', 'CSV');
+```
+
+4. Можно создать таблицу, содержащую все файлы из обоих директорий (все файлы должны соответствовать формату и схеме, описанным в запросе):
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:96]:
```suggestion
Можно создать таблицу, содержащую все файлы из обеих директорий (все файлы должны соответствовать формату и схеме, описанным в запросе):
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+
+``` sql
+CREATE TABLE table_with_asterisk (name String, value UInt32) ENGINE = S3('https://storage.yandexcloud.net/my-test-bucket-768/{some,another}_prefix/*', 'CSV');
+```
+
+!!! warning "Предупреждение"
+    Если список файлов содержит диапазоны чисел с ведущими нулями, используйте конструкцию с фигурными скобками для каждой цифры отдельно или используйте `?`.
+
+**Пример**
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:105]:
```suggestion

```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+
+Создайте таблицу из файлов `file-000.csv`, `file-001.csv`, … , `file-999.csv`:
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:107]:
```suggestion
Создание таблицы из файлов `file-000.csv`, `file-001.csv`, … , `file-999.csv`:
```

Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:107]:
```suggestion
Создание таблицы из файлов `file-000.csv`, `file-001.csv`, … , `file-999.csv`:
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+-   `_path` — путь к файлу.
+-   `_file` — имя файла.
+
+**Смотрите также**
+
+-   [Виртуальные столбцы](index.md#table_engines-virtual_columns)
+
+## Настройки, связанные с S3 {#settings}
+
+Следующие параметры могут быть указаны перед выполнением запроса или в конфигурационном файле:
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:124]:
```suggestion
Перед выполнением запроса или в конфигурационном файле могут быть указаны следующие параметры:
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+
+-   `s3_max_single_part_upload_size` — значение по умолчанию `64Mb`. Максимальный размер объекта для одиночной загрузки в S3.
+-   `s3_min_upload_part_size` — значение по умолчанию `512Mb`. Минимальный размер части файла для [составной загрузки](https://docs.aws.amazon.com/AmazonS3/latest/dev/uploadobjusingmpu.html).
+-   `s3_max_redirects` — значение по умолчанию `10`. Максимальное количество http-перенаправлений в запросе на хранилище S3.
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:128]:
```suggestion
-   `s3_max_single_part_upload_size` — максимальный размер объекта для одиночной загрузки в S3. Значение по умолчанию `64Mb`. 
-   `s3_min_upload_part_size` — минимальный размер части файла для [составной загрузки](https://docs.aws.amazon.com/AmazonS3/latest/dev/uploadobjusingmpu.html). Значение по умолчанию `512Mb`. 
-   `s3_max_redirects` — максимальное количество http-перенаправлений в запросе на хранилище S3. Значение по умолчанию `10`. 
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+
+**Соображения безопасности**: чтобы избежать CSRF атак, когда злоумышленник может указать произвольные URL-адреса S3, следует установить `s3_max_redirects` значение `0`; или обязательно указывать `remote_host_filter` в конфигурации сервера.
+
+## Настройки конечных точек {#endpointsettings}
+
+В файле конфигурации для определенной конечной точки (которая соответствует точному префиксу URL-адреса) можно указать следующие параметры:
+
+-   `endpoint` — обязательный параметр, указывает префикс конечной точки.
+-   `access_key_id` and `secret_access_key` — опциональный параметр. Содержит учетные данные для использования с данной конечной точкой.
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:137]:
```suggestion
-   `access_key_id` и `secret_access_key` — опциональные параметры. Содержат учетные данные для использования с данной конечной точкой.
```

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+-   `use_environment_credentials` — опциональный параметр, значение по умолчанию `false`. Если установлено значение `true`, клиент S3 будет использовать учетные сведения из переменных сред и метаданных Amazon EC2, для данной конечной точки.
+-   `header` — опциональный параметр. Может быть указан несколько раз. Добавляет указанный HTTP-заголовок к запросу к заданной конечной точке.
+-   `server_side_encryption_customer_key_base64` — опциональный параметр. При его использовании, будут установлены необходимые заголовки для доступа к объектам S3 с помощью шифрования SSE-C.
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/s3.md:140]:
```suggestion
-   `endpoint` — указывает префикс конечной точки. Обязательный параметр.
-   `access_key_id` and `secret_access_key` — содержит учетные данные для использования с данной конечной точкой. Опциональный параметр. 
-   `use_environment_credentials` — если установлено значение `true`, клиент S3 будет использовать учетные сведения из переменных сред и метаданных Amazon EC2 для данной конечной точки. Опциональный параметр. Значение по умолчанию `false`. 
-   `header` — добавляет указанный HTTP-заголовок к запросу к заданной конечной точке. Опциональный параметр, может быть указан несколько раз.  
-   `server_side_encryption_customer_key_base64` — устанавливает необходимые заголовки для доступа к объектам S3 с помощью шифрования SSE-C. Опциональный параметр. 
```

File: docs/ru/engines/table-engines/index.md [hunk 18]
```diff
@@ -52,6 +52,7 @@ toc_title: "Введение"
 -   [MySQL](integrations/mysql.md#mysql)
 -   [ODBC](integrations/odbc.md#table-engine-odbc)
 -   [JDBC](integrations/jdbc.md#table-engine-jdbc)
+-   [S3](integrations/s3.md#table_engines-s3)
```
Comment (adevyatova) [inline docs/ru/engines/table-engines/index.md:55]:
```suggestion
-   [S3](integrations/s3.md#table-engine-s3)
```

**Comment by ka1bi4** [comment]
Hi, @excitoon.
Added documentation for s3 in ClickHouse. It's ready for review.

File: docs/en/engines/table-engines/integrations/s3.md [hunk 19]
```diff
@@ -73,13 +73,64 @@ For more information about virtual columns see [here](../../../engines/table-eng
...
+CREATE TABLE table_with_question_mark (name String, value UInt32) ENGINE = S3('https://storage.yandexcloud.net/my-test-bucket-768/{some,another}_prefix/some_file_?', 'CSV');
+```
+
+Table consists of all the files in both directories (all files should satisfy format and schema described in query):
+
+``` sql
+CREATE TABLE table_with_asterisk (name String, value UInt32) ENGINE = S3('https://storage.yandexcloud.net/my-test-bucket-768/{some,another}_prefix/*', 'CSV');
+```
+
+!!! warning "Warning"
```
Comment (alexey-milovidov) [inline docs/en/engines/table-engines/integrations/s3.md:107]: Why?

File: docs/ru/engines/table-engines/integrations/s3.md [hunk 20]
```diff
@@ -73,20 +73,20 @@ SELECT * FROM s3_engine_table LIMIT 2;
...
--   `access_key_id` и `secret_access_key` — указывают учетные данные для использования с данной конечной точкой.
--   `use_environment_credentials` — если `true`, S3-клиент будет пытаться получить учетные данные из переменных среды и метаданных Amazon EC2 для данной конечной точки. Значение по умолчанию - `false`.
--   `header` — добавляет указанный HTTP-заголовок к запросу на заданную конечную точку. Может быть определен несколько раз.
+-   `access_key_id` и `secret_access_key` — указывают учетные данные для использования с данной точкой приема запроса.
+-   `use_environment_credentials` — если `true`, S3-клиент будет пытаться получить учетные данные из переменных среды и метаданных Amazon EC2 для данной точки приема запроса. Значение по умолчанию - `false`.
+-   `header` — добавляет указанный HTTP-заголовок к запросу на заданную точку приема запроса. Может быть определен несколько раз.
 -   `server_side_encryption_customer_key_base64` — устанавливает необходимые заголовки для доступа к объектам S3 с шифрованием SSE-C. 
 
-**Пример**
+Создание таблицы из файлов `file-000.csv`, `file-001.csv`, … , `file-999.csv`:
```
Comment (alexey-milovidov) [inline docs/ru/engines/table-engines/integrations/s3.md:89]: Это про что?

**Comment by ka1bi4** [comment]
Hi @alexey-milovidov 
This PR is ready for merge.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**